src/main/webapp/script.js

```diff
@@ -213,20 +234,300 @@ function getGraphDisplay(graphNodes, graphEdges) {
+
+  // Highlight and retrieve the objects that will be modified by this mutation
+  const result = highlightDiff(cy, mutList, reason);
+  if (Object.keys(result).length === 0) {
+    return;
+  }
+  // Break the list down into individual constituents
+  const deletedNodes = result["deletedNodes"] || cy.collection();
+  const deletedEdges = result["deletedEdges"] || cy.collection();
+  const addedNodes = result["addedNodes"] || cy.collection();
+  const addedEdges = result["addedEdges"] || cy.collection();
+  const modifiedNodes = result["modifiedNodes"] || cy.collection();
+
+  let elems = cy.collection();
+  Object.entries(result).forEach(([, value]) => { elems = elems.union(value); })
+
+  // Reposition added elements
+  cy.layout({
+    name: 'dagre'
+  }).run();
+
+  // Zoom in on them and activate their reason tooltips
+  makeInteractiveAndFocus(cy, elems);
+
+
+  showMutButton.addEventListener("change", () => {
+    if (showMutButton.checked) {
+      /*
+       * We take advantage of the efficiency of batch operations to avoid calling
+       * highlightDiff again each time the checkbox is clicked
+       */
+      showDiffs(cy, elems, deletedNodes, deletedEdges, addedNodes, addedEdges, modifiedNodes);
+    } else {
+      hideDiffs(cy, elems, deletedNodes, deletedEdges, addedNodes, addedEdges, modifiedNodes);
+    }
+  });
+  return cy;
+}
+
+
+/**
+ * Highlights modified nodes and edges in the graph according to the list
+ * of mutations
+ * 
+ * @param {*} cy the graph 
+ * @param {*} mutList the list of mutations to highlight
+ * @param {*} reason the reason for the mutations
+ * @returns {*} an object containing the deleted nodes, deleted edges, added
+ * nodes, added edges and modified nodes as per the mutationList or an empty
+ * object if there are no mutations
+ */
+function highlightDiff(cy, mutList, reason = "") {
+  // Initialize empty collections
+  let deletedNodes = cy.collection();
+  let deletedEdges = cy.collection();
+  let addedNodes = cy.collection();
+  let addedEdges = cy.collection();
+  let modifiedNodes = cy.collection();
+
+  // If the mutation list is null/undefined
+  if (!mutList) {
+    return {};
+  }
+  // Apply each mutation
+  mutList.forEach(mutation => {
+    const type = mutation["type_"] || -1;
+    const startNode = mutation["startNode_"];
+    const endNode = mutation["endNode_"];
+    let modifiedObj = cy.collection();
+
+    if (!type || !startNode) {
+      return modifiedObj;
+    }
+
+    switch (type) {
+      case 1:
+        // add node
+        if (cy.getElementById(startNode).length !== 0) {
+          modifiedObj = cy.getElementById(startNode);
+          // color this node green
+          modifiedObj.style('background-color', colorScheme["addedObjectColor"]);
+          addedNodes = addedNodes.union(modifiedObj);
+        }
+        break;
+      case 2:
+        // add edge
+        if (endNode && cy.getElementById(startNode).length !== 0 && cy.getElementById(endNode).length !== 0) {
```

Comment:
You might want to use the alias for `cy.getElementById()` which is `cy.$id()` at least on this line because it is getting pretty long. You might just want to use it everywhere else though because it is a lot shorter.
https://js.cytoscape.org/#cy.getElementById